src/browser/FileTree.js

```diff
@@ -755,5 +548,36 @@ class FileTree extends LitElement {
       cueEl.classList.add("loading-cue-overlay--slide");
     }
   }
+
+  async downloadAsZIP() {
+    const zipWriter = new zip.ZipWriter(new zip.BlobWriter("application/zip"));
+
+    // Add all files to zip
+    const allFiles = await getAllFiles();
+    await Promise.all(
+      Object.entries(allFiles).map(
+        ([key, value]) =>
+          new Promise(async (resolve) => {
```

Comment:
why not just returning `zipWriter.add(key, new zip.TextReader(value));`
It's already a Promise I think.